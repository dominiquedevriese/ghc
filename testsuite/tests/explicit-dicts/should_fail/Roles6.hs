{-# LANGUAGE FlexibleContexts #-}
{-# LANGUAGE TypeApplications #-}
module Roles6 where

import GHC.Base (Ord(..))

reverseOrd :: Ord a => Ord.Dict a
reverseOrd = undefined

f6 = (undefined :: Ord Int => Int -> [Int] -> [Int])
     ((reverseOrd @Int :: Ord Int))
-- Flagged by coherence check
