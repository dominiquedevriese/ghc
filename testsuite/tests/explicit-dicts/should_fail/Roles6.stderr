
Roles6.hs:10:6: error:
    • Explicit dictionary application to: ‘Ord Int’
      is not allowed because the instance incoherence check
      requires type-variable arguments
    • In the expression:
        (undefined :: Ord Int => Int -> [Int] -> [Int])
          ((reverseOrd @Int :: Ord Int))
      In an equation for ‘f6’:
          f6
            = (undefined :: Ord Int => Int -> [Int] -> [Int])
                ((reverseOrd @Int :: Ord Int))
